new_platforms/cmake/add_enclave.cmake

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+function(add_enclave)
+    # Borrowed from ../cmake/add_enclave.cmake
+    # Using the same signature so that the functions are easier to merge.
+    set(options CXX)
+    set(oneValueArgs TARGET UUID CONFIG KEY)
+    set(multiValueArgs SOURCES C_GEN)
+    cmake_parse_arguments(ENCLAVE
+        "${options}"
+        "${oneValueArgs}"
+        "${multiValueArgs}"
+        ${ARGN})
+
+    if(TZ)
+        set(ENCLAVE_TARGET ${ENCLAVE_UUID})
+    endif()
+
+    if(UNIX AND TZ)
+        add_custom_target(${ENCLAVE_TARGET} ALL
+            COMMAND make -C ${CMAKE_CURRENT_SOURCE_DIR}/optee
+                -f linux_gcc.mak
+                OE_INC=${OE_PATH}/include
+                NP_INC=${NP_PATH}/include
+                OPTEE_OS_PATH=${OPTEE_OS_PATH}
+                CYREP_PATH=${CYREP_PATH}
+                MBEDTLS_PATH=${MBEDTLS_PATH}
+                TINYCBOR_PATH=${TINYCBOR_PATH}
+                TA_DEV_KIT_DIR=${TA_DEV_KIT_DIR}
+                CROSS_COMPILE=${OE_TA_TOOLCHAIN_PREFIX}
+                O=${CMAKE_CURRENT_BINARY_DIR}
+                AR_O=${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}
+                GEN=${ENCLAVE_C_GEN}
+            DEPENDS ${ENCLAVE_SOURCES}
+            SOURCES ${ENCLAVE_SOURCES})
+            add_dependencies(${ENCLAVE_TARGET} oeenclave liboestdio_enc liboesocket_enc)
+            add_custom_command(TARGET ${ENCLAVE_TARGET} POST_BUILD
+                COMMAND ${CMAKE_COMMAND} -E
+                    copy ${CMAKE_CURRENT_BINARY_DIR}/*.ta ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
+    elseif(WIN32 AND (SGX OR (TZ AND SIM)))
+        add_library(${ENCLAVE_TARGET} MODULE ${ENCLAVE_SOURCES})
+
+        if(SGX)
+            target_compile_definitions(${ENCLAVE_TARGET} PUBLIC OE_USE_SGX)
```

Comment:
I'll remove `OE_USE_SGX` and `OE_USE_OPTEE` from here once `user/hegatta/nuget` gets merged into `feature.new_platforms`.